core/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -139,4 +140,14 @@
      */
     boolean readOnly();
 
+    /**
+     * Resolves snapshot name(s) in the repository to {@link SnapshotId}(s).  If any snapshots
+     * are not found, a {@link SnapshotMissingException} is thrown.
+     *
+     * @param snapshotNames snapshot name(s)
+     * @return snapshot ids
+     * @throws SnapshotMissingException if a snapshot is not found
+     */
+    List<SnapshotId> resolveSnapshots(String... snapshotNames);
```

Comment:
maybe call this "resolveSnapshotNames"?
I would also prefer to use `List<String> snapshotNames` as parameter to bring it closer to the return type.